my-app/src/state/state.js

```diff
@@ -4,19 +4,55 @@ let state = {
   functions: {},
 
   data: {
-    zonesList: {},
-
-    cardsList: {
-      id: 1,
-      isDone: false,
-      task: "html + css",
-      cadrId: 1,
-      colormark: "green",
-    },
+    zonesList: [
```

Comment:
Just a few thoughts about data structure. It will be more important when you will think about database integration.
There are a lot of different ways to represent data and their relations. They have pros and cons, and it's up to you to decide which approach will better fit your particular problem. 

The most straightforward approach is called data normalization. It is used mainly in relational databases (aka SQL databases: MySQL, Postgres, etc.). In this approach, you think of data as multiple tables which contains entities with links to the related entities. For instance, in this particular case, the data structure could look like:

```
{
  zones: [
    { id: 1, name: 'todo' },
    { id: 2, name: 'in progress' },
     { id: 3, name: 'in review' },
     { id: 4, name: 'done' },
  ],
  cards: [
    { id: 1, name: 'do stuff', zoneId: 1 }, // it means "do stuff" is in 'todo' zone
    { id: 2, name: 'do other stuff', zoneId: 1 },
    { id: 3, name: 'do another stuff', zoneId: 4 },
  ]
}
```

If you would like to add assignees for the cards in the future, data structure will be changed to the following:
```
{
  zones: [
    { id: 1, name: 'todo' },
    { id: 2, name: 'in progress' },
     { id: 3, name: 'in review' },
     { id: 4, name: 'done' },
  ],
  cards: [
    { id: 1, name: 'do stuff', zoneId: 1, assignedTo: 1  }, // card assigned to 'user 1' 
    { id: 2, name: 'do other stuff', zoneId: 1, assignedTo: null }, // card is not assigned
    { id: 3, name: 'do another stuff', zoneId: 4 },
  ],
  users: [
    { id: 1, name: 'user 1' },
    { id: 1, name: 'user 2' },
  ]
}
```



The biggest advantage of this approach - you can combine data in any manner you want. You can easily find which tasks are assigned to whom, or who has done the biggest number of tasks, etc.
The biggest disadvantage - you need to combine them before use. For large data sets that would be time-consuming.

What you are doing now is called denormalized data representation. This approach is used mostly in NoSQL databases, like dynamodb, elasticsearch, and sometimes mongo. In this approach, you create a data structure based on how you will use it. And it could be pretty tricky, as you should know the use cases you will have in the application in advance. But the advantage is performance. If you model data properly, performance will be like 0(1) for most cases. This application is a great example - you already have the data structure you need for the application. You do not need to group anything or find relations. But if you would like to find "all tasks assigned to a specific user" it will be more complex than with normalized data.


I would recommend using normalized data representation as it is pretty straightforward and it is nice to know how to do that. And proper denormalization will be much easier when you know how to keep data normalized.